source/Camera/InfoUtilOther/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,102 @@
+	const HalconCpp::HString filetype{ "ply" };
+	const HalconCpp::HString filenameCstring{ filename.c_str() };
+	const HalconCpp::HString genParamName{ "invert_normals" };
+	const HalconCpp::HString genParamValue{ "false" };
+	model.WriteObjectModel3d(filetype, filenameCstring, genParamName, genParamValue);
+}
+
+
+HalconCpp::HObjectModel3D setColorInObjectModel3D(HalconCpp::HImage RGB, HalconCpp::HObjectModel3D pcHalcon) {
+
+	HalconCpp::HImage R = RGB.AccessChannel(1);
+	HalconCpp::HImage G = RGB.AccessChannel(2);
+	HalconCpp::HImage B = RGB.AccessChannel(3);
+
+	HalconCpp::HRegion domain = R.GetDomain();
+	HalconCpp::HTuple rows, cols;
+	domain.GetRegionPoints(&rows, &cols);
+
+	HalconCpp::HTuple Rtup = R.GetGrayval(rows, cols);
+	HalconCpp::HTuple Gtup = G.GetGrayval(rows, cols);
+	HalconCpp::HTuple Btup = B.GetGrayval(rows, cols);
+
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+
+	return pcHalcon;
+}
+
+
+int main()
+{
+	try
+	{
+		std::cout << "Connecting to camera" << std::endl;
+
+		// Connect to camera using GeniCamTL driver
+		HalconCpp::HFramegrabber Framegrabber;
+		Framegrabber.OpenFramegrabber("GenICamTL", 1, 1, 0, 0, 0, 0, "progressive", -1, "default", -1, "false", "default", "default", 0, 0);
+
+		// Configure 3D-settings
+		Framegrabber.SetFramegrabberParam("create_objectmodel3d", "enable");
+		Framegrabber.SetFramegrabberParam("add_objectmodel3d_overlay_attrib", "enable");
+		Framegrabber.SetFramegrabberParam("AcquisitionMode", "SingleFrame");
+
+		// Set camera settings
+		Framegrabber.SetFramegrabberParam("Iris", 20);
+		Framegrabber.SetFramegrabberParam("ExposureTime", 8333);
+		Framegrabber.SetFramegrabberParam("Gain", 2);
+		Framegrabber.SetFramegrabberParam("Brightness", 1.0);
+
+		Framegrabber.SetFramegrabberParam("OutlierFilterEnabled", 1);
+		Framegrabber.SetFramegrabberParam("GaussianFilterEnabled", 1);
+		Framegrabber.SetFramegrabberParam("GaussianFilterSigma", 1.5);
+		Framegrabber.SetFramegrabberParam("OutlierFilterThreshold", 5);
+
+		// Capture frame
+		std::cout << "Capturing frame" << std::endl;
+
+		HalconCpp::HRegion region;
+		HalconCpp::HXLDCont contours;
+		HalconCpp::HTuple data;
+		HalconCpp::HImage frame = Framegrabber.GrabData(&region, &contours, &data);
+
+		// Construct HObjectModel3D from x,y,z data
+		HalconCpp::HImage X = frame.SelectObj(1);
+		HalconCpp::HImage Y = frame.SelectObj(2);
+		HalconCpp::HImage Z = frame.SelectObj(3);
+		HalconCpp::HImage RGB = frame.SelectObj(5);
+
+		HalconCpp::HObjectModel3D pcHalcon;
+		pcHalcon.XyzToObjectModel3d(X, Y, Z);
+
+		// Color the object based on R,G,B data
+		pcHalcon = setColorInObjectModel3D(RGB, pcHalcon);
+
+		// Delete zero-points (not set due to nans in the Zivid point cloud)
+		pcHalcon = pcHalcon.SelectPointsObjectModel3d("point_coord_z", 0.1, 15000);
```

Comment:
Does this create a new pcHalcon object? If so, probably don't want to reuse the same variable. It would be better if there is a function that just modifies the existing objectModel3D?